src/routers/alpha-router/alpha-router.ts

```diff
@@ -1737,6 +1752,43 @@ export class AlphaRouter
       return Promise.resolve(null);
     }
 
+    if (v4Routes.length > 0) {
+      const v4RoutesFromCache: V4Route[] = v4Routes.map(
+        (cachedRoute) => cachedRoute.route as V4Route
+      );
+      metric.putMetric(
+        'SwapRouteFromCache_V4_GetQuotes_Request',
+        1,
+        MetricLoggerUnit.Count
+      );
+
+      const beforeGetQuotes = Date.now();
+
+      quotePromises.push(
+        this.v4Quoter
+          .getQuotes(
+            v4RoutesFromCache,
+            amounts,
+            percents,
+            quoteToken,
+            tradeType,
+            routingConfig,
+            undefined,
+            v3GasModel,
```

Comment:
is this the same as v3 or will it change?

Reply:
It will need to change. I have a https://linear.app/uniswap/issue/ROUTE-223/support-v4-gas-model ticket.